resources/src/jvmMain/kotlin/dev/icerock/moko/resources/ImageResource.kt

```diff
@@ -15,6 +22,35 @@ actual class ImageResource(
     val image: BufferedImage by lazy {
         val stream = resourcesClassLoader.getResourceAsStream(filePath)
             ?: throw FileNotFoundException("Couldn't open resource as stream at: $filePath")
-        stream.use { ImageIO.read(it) }
+        stream.use {
+            if (filePath.endsWith(".svg", ignoreCase = true)) {
+                readSvg(it)
+            } else {
+                ImageIO.read(it)
+            }
+        }
+    }
+
+    private fun readSvg(
+        inputStream: InputStream
+    ): BufferedImage {
+        // Create a PNG transcoder.
+        val t: Transcoder = PNGTranscoder()
+
+        // Create the transcoder input.
+        val input = TranscoderInput(inputStream)
+
+        // Create the transcoder output.
+        val outputStream = ByteArrayOutputStream()
+        outputStream.use {
+            val output = TranscoderOutput(it)
+
+            // Save the image.
+            t.transcode(input, output)
+        }
+
+        // Convert the byte stream into an image.
+        val imgData: ByteArray = outputStream.toByteArray()
+        return ImageIO.read(ByteArrayInputStream(imgData))
```

Comment:
i see that here created temporary ByteArray that can be removed.
can we use `PipedOutputStream` and `PipedInputStream` instead of `ByteArray` streams?